### DISCUSSION TITLE
5 giugno 2023 - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#39] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/0e936523-bbef-46a8-b3bc-62bbc3c2cb70)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/4d886233-1f20-422e-ae21-7a77e89a863f)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/cca2f777-9e30-4302-b24a-befd0d37b664)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/e3f6661c-5ad2-438f-afde-4b57223ba5e4)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/f9dcfee0-6709-4933-8f38-a20a15f542dc)
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/f47b5316-299b-458b-8a40-836d84e87e7c)

**Reply by Alegau03** [reply]
Hai sbagliato la tabella hai messo b1 in ABC, infatti a me viene che ha join senza perdita
<img width="732" alt="image" src="https://github.com/sapienzastudentsnetwork/bdi2324/assets/113530268/04343d38-2189-4271-885d-e2da1d307390">

**Reply by Warcophyr** [reply]
Versione corretta
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/d808f5e4-a8bf-4f7c-bfd9-0585151639df)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-14 13:00:54](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/b366a148-ccdd-4c1a-a5b3-30aeb3d63583)
![IMAGE 2024-01-14 13:01:01](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/744c5b32-050d-483b-b78a-25fe494c4aef)

**Comment by Frisbon** [comment]
2a) Controllo le unioni delle dipendenze e vedo se sono in $\rho$. If true, le skippo.

BDA e DAC stanno in ABCD, EDAC ha una mamma e tanti papá

- ED→AC ⇒ Calcolo $ED^+_G$  e vedo se AC ne fa parte…

Z = ED, S = {}

1st cycle, per ogni sottoschema… (Calcolo diretto)
$S= S \cup{((ED\cap \rho_i)^+_F \cap \rho_i )}$

quindi
$S= ∅ \cup{((ED\cap ABCD)^+_F \cap ABCD )} \cup{((ED\cap ADE)^+_F \cap ADE )}$ 

$S= {((D)^+_F \cap ABCD )} \cup{((ED)^+_F \cap ADE )}$ 

$S= {(D \cap ABCD )} \cup{(ACDE \cap ADE )} = D \cup ADE = ADE$

Se S in Z stop, false ⇒ Z = Z+S and cycle again…

2nd cycle…

$S= ADE \cup{((ADE\cap ABCD)^+_F \cap ABCD )} \cup{((ADE\cap ADE)^+_F \cap ADE )}$ 

$= ADE \cup (AD^+_F \cap ABCD) \cup ADE$

$= ADE \cup ADC \cup ADE = ACDE$

3rd cycle…

$S = ACDE \cup{((ACDE\cap ABCD)^+_F \cap ABCD )} \cup{((ACDE\cap ADE)^+_F \cap ADE )}$ 

= ACDE senza calcolare perché sono pigro e tiro a caso

quindi

$ED^+_G = ACDE$ della quale fa parte $AC$  ⇒ **F preservativato.**

---
### Per sbaglio ho calcolato se viene preservato, non il lossless join, un esercizio che nobody asked
2b) ~~BAC~~, AG, GC, CGB (KGB?????? no way блять)

~~Il prof é stronzo e ci fa calcolare tre chiusure di G…~~ Se tutte e tre sono verificate (ossia il det.ato appartiene) allora F preservato.

- $A^+_G$

$S= ∅ \cup{((A\cap ABC)^+_F \cap ABC )} \cup{((A\cap BEG)^+_F \cap BEG )} \cup{((A\cap ADE)^+_F \cap ADE )}$  

= ABC che not in A, riparto con Z = ABC

$S= ABC \cup{((ABC\cap ABC)^+_F \cap ABC )} \cup{((ABC\cap BEG)^+_F \cap BEG )} \cup{((ABC\cap ADE)^+_F \cap ADE )}$ = ABCG che not in Z riparto

… (non mi va)

$A^+_G = ABCG$
 e $G$ inside, ok 

- $G^+_G$

$S= ∅ \cup{((G\cap ABC)^+_F \cap ABC )} \cup{((G\cap BEG)^+_F \cap BEG )} \cup{((G\cap ADE)^+_F \cap ADE )}$   = BG, restart

$S= BG \cup{((BG\cap ABC)^+_F \cap ABC )} \cup{((BG\cap BEG)^+_F \cap BEG )} \cup{((BG\cap ADE)^+_F \cap ADE )}$  = ABCG, restart

… (non mi va)

$G^+_G = ABCG$
 e $C$ inside, nice

- $CG^+_G$

$S= ∅ \cup{((CG\cap ABC)^+_F \cap ABC )} \cup{((CG\cap BEG)^+_F \cap BEG )} \cup{((CG\cap ADE)^+_F \cap ADE )}$  = BG, restart

$S= BCG \cup{((BCG\cap ABC)^+_F \cap ABC )} \cup{((BCG\cap BEG)^+_F \cap BEG )} \cup{((BCG\cap ADE)^+_F \cap ADE )}$  = ABCG, restart

… (yet ancora non mi va)

$CG^+_G = ABCG$ e $B$ inside, nice

**Siccome tutte e tre verificate ⇒ F preservato**
